### PR TITLE
Add _cat_file

### PR DESCRIPTION
**Pull request by kephale**
Howdy folks, sshfs is amazing. Thank you for everything!

I was trying to use it for remote visualization of [zarr](https://zarr.readthedocs.io/en/stable/) images. This required support for `_cat_file`. This PR provides a `_cat_file` implementation.

### CODE REVIEW

File: sshfs/spec.py [hunk 1]
```diff
@@ -316,3 +316,12 @@ async def _execute(self, *args, **kwargs):
...
         return SSHFile(self, path, *args, **kwargs)
+
+    @wrap_exceptions
+    async def _cat_file(self, path, **kwargs):
+        """Asynchronously fetch the contents of a file"""
+        async with self._pool.get() as channel:
+            async with channel.open(path, "rb") as f:
+                return await f.read()
+
+    cat_file = sync_wrapper(_cat_file)
```
Comment (shcheklein) [inline sshfs/spec.py:327]: I think it's not needed. 

From the [docs](https://filesystem-spec.readthedocs.io/en/latest/async.html#implementing-new-backends):

>  These functions will either have sync versions automatically generated is the name is in the async_methods list, or can be directly created using sync_wrapper.

PTAL https://github.com/fsspec/filesystem_spec/blob/master/fsspec/asyn.py#L275 ... and https://github.com/fsspec/filesystem_spec/blob/master/fsspec/asyn.py#L913

File: sshfs/spec.py [hunk 2]
```diff
@@ -316,3 +316,12 @@ async def _execute(self, *args, **kwargs):
 
     def _open(self, path, *args, **kwargs):
         return SSHFile(self, path, *args, **kwargs)
+
+    @wrap_exceptions
+    async def _cat_file(self, path, **kwargs):
```
Comment (shcheklein) [inline sshfs/spec.py:321]: let's try to add a test for this please

**Review by shcheklein — COMMENTED** [review]
Thanks @kephale ! Looks good, I think someone else will also take a look and merge. A few comments. Please take a look.

File: sshfs/spec.py (hunk 1 above)
Comment (kephale) [reply sshfs/spec.py]: Aha, got it. I removed the sync_wrapper call.

File: sshfs/spec.py (hunk 2 above)
Comment (kephale) [reply sshfs/spec.py:321]: Test added!

**Review by shcheklein — APPROVED** [review]

**Comment by shcheklein** [comment]
@martindurant @efiop PTAL and merge if it looks good (I don't have access to this repo).

**Comment by martindurant** [comment]
LGTM

@efiop , what are your plans for maintenance of this repo? Are you still available for making releases?

**Comment by efiop** [comment]
@martindurant I've set up everything in the past to work even if I get hit by a bus :) Just for the record, one could just go to https://github.com/fsspec/sshfs/releases and click release, gha will do its magic 🙂 Triggered 2024.4.0 for now. As to maintaining, I am keeping an eye, but obviously no promises on the reaction time.

**Comment by efiop** [comment]
Also added @shcheklein to admins here.

**Pull request merged**